common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/uplink/DefaultLwM2mUplinkMsgHandler.java

```diff
@@ -501,16 +512,27 @@ private void sendReadRequests(LwM2mClient lwM2MClient, Lwm2mDeviceProfileTranspo
         }
     }
 
-    private void sendObserveRequests(LwM2mClient lwM2MClient, Lwm2mDeviceProfileTransportConfiguration profile, Set<String> supportedObjects) {
+    private void sendInitObserveRequests(LwM2mClient lwM2MClient, Lwm2mDeviceProfileTransportConfiguration profile, Set<String> supportedObjects) {
         try {
             Set<String> targetIds = profile.getObserveAttr().getObserve();
             targetIds = targetIds.stream().filter(target -> isSupportedTargetId(supportedObjects, target)).collect(Collectors.toSet());
-
-            CountDownLatch latch = new CountDownLatch(targetIds.size());
-            targetIds.forEach(targetId -> sendObserveRequest(lwM2MClient, targetId,
-                    new TbLwM2MLatchCallback<>(latch, new TbLwM2MObserveCallback(this, logService, lwM2MClient, targetId))));
-
-            latch.await(config.getTimeout(), TimeUnit.MILLISECONDS);
+            if (!targetIds.isEmpty()) {
+                TelemetryObserveStrategy observeStrategy = profile.getObserveAttr().getObserveStrategy();
+                if (SINGLE.equals(observeStrategy)) {
+                    CountDownLatch latch = new CountDownLatch(targetIds.size());
+                    targetIds.forEach(targetId -> sendObserveRequest(lwM2MClient, targetId,
+                            new TbLwM2MLatchCallback<>(latch, new TbLwM2MObserveCallback(this, logService, lwM2MClient, targetId))));
+                    latch.await(config.getTimeout(), TimeUnit.MILLISECONDS);
+                } else if (COMPOSITE_ALL.equals(observeStrategy)) {
+                    String[] versionedIds = targetIds.toArray(new String[0]);
```

Comment:
add latch just in case to avoid race condition with sendWriteAttributeRequests.

Reply:
ok